public/scripts/week-calendar.js

```diff
@@ -1069,18 +1069,28 @@ function checkNotesValidations(form) {
     notes_tag_error.innerText = "tag cant be longer than 20 characters";
     form.tag.setCustomValidity("error");
   }
-  if (!form.tag.value.match(/^[a-zA-Z]+$/)) {
+  if (
+    typeof form.tag.value === "string" &&
+    form.tag.value.trim().length > 0 &&
+    !form.tag.value.match(/^[a-zA-Z0-9_]+$/)
```

Comment:
the regex needs to be present everywhere tag is validated ( currently its missing in alot of places ) 
![image](https://user-images.githubusercontent.com/91623753/236294270-33d8a988-9e39-435b-ae67-4fa797ecc305.png)